app/code/Magento/Catalog/Block/Product/View/Gallery.php

```diff
@@ -170,10 +171,110 @@ public function getGalleryImagesJson()
         return json_encode($imagesItems);
     }
 
+    /**
+     * Retrieve gallery options in JSON format
+     *
+     * @return string
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+     * @SuppressWarnings(PHPMD.NPathComplexity)
+     */
+    public function getGalleryOptionsJson()
+    {
+        $optionItems = null;
+
+        //Special case for gallery/nav which can be the string "thumbs/false/dots"
+        is_bool($this->getVar("gallery/nav")) ?
+            $optionItems['nav'] = $this->getVar("gallery/nav") ? 'true' : 'false' :
+            $optionItems['nav'] = $this->escapeHtml($this->getVar("gallery/nav"));
+
+        $optionItems['loop'] = $this->getVar("gallery/loop");
```

Comment:
According to the documentation, the value of gallery/loop should be <true/false>, i.e a boolean not a string.

The previous way of doing it, i.e `$block->getVar("gallery/loop") ? 'true' : 'false'` outputs a string 'true' or 'false' which is not correct.

Because of the changes made in resolution to #12285 the getVar function will now return

boolean true if gallery/loop existed and was set to true
boolean false if gallery/loop existed and was set to false
boolean false if gallery/loop doesn't exist (Arghhh, this is why we need to relook at #12285, a missing variable doesn't necessarily mean that it should be set to false by default!)

So, passing the boolean result of the getVar function straight for JSON encoding will result in the correct output, i.e <true/false> and not <"true"/"false">

The same applies for the remaining booleans.